File: domoticz/script/perl/domoticz.pl
#!/usr/bin/env perl
   use v5.14;
   use LWP::Simple;                # From CPAN
   use Data::Dumper;               # Perl core module
   use strict;                     # Good practice
   use warnings;                   # Good practice
   use utf8;
   use Switch;
   use Config::Simple;
   use feature     qw< unicode_strings >;
   
################################################

   
	# Configuration section, please update to your values
   my $conf = "/home/pi/domoticz/scripts/config.properties";
   my $cfg;
   my $host;
   my $idx;
   my $cmd;
   my $Config=&read_conf($conf);
   my $ConfigIdx=&read_conf("/home/pi/domoticz/scripts/domoticz.properties");
   
   # ip and port of your Domoticz server 
   my $url = $Config->{"default.domoticz.url"}; 
   my $command=$ARGV[0];
   
switch ( $command ) {
	case "volume_up" {
		my $idx_vol_up = $ConfigIdx->{"switchs.idx.D_AMPLI_VUP"}; 
		#print $url.'type=command&param=switchlight&idx='.$idx_vol_up.'&switchcmd=On';
		setSwitch($url,$idx_vol_up,"On");
	}
	case "volume_down" {
		my $idx_vol_down = $ConfigIdx->{"switchs.idx.D_AMPLI_VDOWN"};
		setSwitch($url,$idx_vol_down,"On");
	}
	case "mute_on" {
		my $idx_vol_mute = $ConfigIdx->{"switchs.idx.D_AMPLI_MUTE"}; 
		setSwitch($url,$idx_vol_mute,"On");
	}
	# update variable
	case "source" {
		my $idx_v_source = $ConfigIdx->{"switchs.idx.V_SOURCE"}; 
		setSwitch($url,$idx_v_source,"On");
	}
	case "v_volume_up" {
		my $idx_v_volume_up = $ConfigIdx->{"switchs.idx.V_VOLUMEUP"};  
		 setSwitch($url,$idx_v_volume_up,"On");
	}
	case "v_volume_down" {
		my $idx_v_volume_down = $ConfigIdx->{"switchs.idx.V_VOLUMEDOWN"};
		setSwitch($url,$idx_v_volume_down,"On");
	}
	case "v_mute_on" {
		my $idx_v_mute = $ConfigIdx->{"switchs.idx.V_MUTE"}; 
		setSwitch($url,$idx_v_mute,"On");
	}
}


 sub setSwitch {
	my $host=$_[0];
	my $idx=$_[1];
	my $cmd=$_[2];
	get($host."type=command&param=switchlight&idx=".$idx."&switchcmd=".$cmd);
 
 }

 sub read_conf {
	my $conf=$_[0];
	$cfg = new Config::Simple($conf) or die Config::Simple->error();
	$cfg->autosave(1);
	# getting the values as a hash:
	#print $cfg->vars();
	my %Config = $cfg->vars();
	return \%Config;
}

File: domoticz/script/python/domoticz.py
#!/usr/bin/env python

import time
import sys 
import os
sys.path.append(os.path.abspath("/home/pi/domoticz/scripts"))
from domoticz_api import *


idx_vol_mute = config.get('domoticz', 'idx.mute');
idx_vol_up = config.get('domoticz', 'idx.vol.up');
idx_vol_down = config.get('domoticz', 'idx.vol.down');
idx_v_mute = config.get('domoticz', 'idx.v_mute');
idx_v_source = config.get('domoticz', 'idx.v_source');
idx_v_volume_up = config.get('domoticz', 'idx.v_volume.up');
idx_v_volume_down = config.get('domoticz', 'idx.v_volume.down');

command = sys.argv[1]

if command=='volume_up':
     set_state_idx(idx_vol_up,'On')
elif command=='volume_down':
     set_state_idx(idx_vol_down,'On')
elif command=='mute_on':
     set_state_idx(idx_vol_mute,'On')
elif command=='source':
     set_state_idx(idx_v_source,'On')
elif command=='v_volume_up':
     set_state_idx(idx_v_volume_up,'On')
elif command=='v_volume_down':
     set_state_idx(idx_v_volume_down,'On')
elif command=='v_mute_on':
     set_state_idx(idx_v_mute,'On')
